code/game/objects/items/airbubble.dm

```diff
@@ -0,0 +1,462 @@
+	//Now it actually works, also locker breakout time stacks with locking and welding
+	//This means secure lockers are more useful for imprisoning people
+	var/breakout_time = 1 * req_breakout()//1.5 minutes if locked or welded, 3 minutes if both
+	if(breakout_time <= 0)
+		return
+
+
+
+	//okay, so the closet is either welded or locked... resist!!!
+	escapee.next_move = world.time + 100
+	escapee.last_special = world.time + 100
+	escapee << "<span class='warning'>You lean on the back of \the [src] and start punching internal wall with your legs. (this will take about [breakout_time] minutes)</span>"
+	visible_message("<span class='danger'>\The [src] begins to shake violently! Something is terring it from the inside!</span>")
+
+	var/time = 6 * breakout_time * 2
+
+	var/datum/progressbar/bar
+	if (escapee.client && escapee.client.prefs.toggles_secondary & PROGRESS_BARS)
+		bar = new(escapee, time, src)
+
+	breakout = 1
+	for(var/i in 1 to time) //minutes * 6 * 5seconds * 2
+		playsound(loc, "sound/items/[pick("rip1","rip2")].ogg", 100, 1)
+		animate_shake()
+
+		if (bar)
+			bar.update(i)
+
+		if(!do_after(escapee, 50, display_progress = FALSE)) //5 seconds
+			breakout = 0
+			qdel(bar)
+			return
+
+		if(!escapee || escapee.stat || escapee.loc != src)
+			breakout = 0
+			qdel(bar)
+			return //closet/user destroyed OR user dead/unconcious OR user no longer in closet OR closet opened
+
+		//Perform the same set of checks as above for weld and lock status to determine if there is even still a point in 'resisting'...
+		if(!req_breakout())
+			breakout = 0
+			qdel(bar)
+			return
+
+	//Well then break it!
+	breakout = 0
+	escapee << "<span class='warning'>You successfully break out! Tearing the bubble's walls!</span>"
+	visible_message("<span class='danger'>\the [escapee] successfully broke out of \the [src]! Tearing the bubble's walls!</span>")
+	playsound(loc, "sound/items/[pick("rip1","rip2")].ogg", 100, 1)
+	break_open()
+	animate_shake()
+	qdel(bar)
+
+/obj/structure/closet/airbubble/break_open()
+	ripped = TRUE
+	update_icon()
+	dump_contents()
+
+/obj/structure/closet/airbubble/verb/set_internals(mob/user as mob)
+	set src in oview(1)
+	set category = "Object"
+	set name = "Set Internals"
+	if(!isnull(internal_tank))
+		visible_message("<span class='warning'>[user] is setting [src] internals.</span>")
+		user << "<span class='notice'>You are settting [src] internals.</span>"
+		if (!do_after(user, 2 SECONDS, act_target = src))
+			return
+		visible_message("<span class='warning'>[user] have set [src] internals.</span>")
+		user << "<span class='notice'>You set [src] internals.</span>"
+		if(use_internal_tank)
+			STOP_PROCESSING(SSfast_process, src)
+		else
+			START_PROCESSING(SSfast_process, src)
+		use_internal_tank = !use_internal_tank
+	else
+		visible_message("<span class='notice'>[src] has no internal tank.</span>")
+
+/obj/structure/closet/airbubble/verb/take_tank(mob/user as mob)
+	if(use_internal_tank)
+		set src in oview(1)
```

Comment:
This form of syntax is generally not used.  The `set` directives are almost always int he header.